### PR TITLE
update tendermint config

### LINKED ISSUE #896: Tendermint config is outdated
I'm trying to use the config module with Tendermint v0.34.x and I found that lots of the options have been changed, so the config fails to parse.

### PR DESCRIPTION
**Pull request by tzemanovic**
closes #896

Hello, I'm trying to use the config module with Tendermint v0.34.x and I found that lots of the options have been changed. I've updated them to be in line with https://github.com/tendermint/tendermint/blob/v0.34.10/config/toml.go.

* [x] Referenced an issue explaining the need for the change
* [x] Updated all relevant documentation in docs
* [x] Updated all code comments where relevant
* [x] Wrote tests
* [x] Added entry in `.changelog/`

### CODE REVIEW

**Review by tarcieri — DISMISSED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 - `[tendermint-abci,tendermint-rpc]` Fix DeliverTx response deserialization
   issues with `gas_wanted` and `gas_used` fields ([#876])
+- `[tendermint]` Update TendermintConfig for Tendermint v.0.34.x ([#896])
```
Comment (thanethomson) [inline CHANGELOG.md:7]: Please add this entry as a separate file in the `.changelog/unreleased/bug-fixes` folder.

See <https://github.com/informalsystems/tendermint-rs/blob/master/CONTRIBUTING.md#changelog> for details.

**Review by thanethomson — CHANGES_REQUESTED** [review]
This looks great, thanks very much! :tada: :pray: Just one minor request regarding the changelog.

File: CHANGELOG.md (hunk 1 above)
Comment (tzemanovic) [reply CHANGELOG.md]: Thank you too,`unclog` looks very handy for this! 

Just a note, the format shown in CONTRIBUTING shows `- [pkg] \#xxx Some description about the change`, but the other entries in `.changelog` had a slightly different format, so I followed that for consistency.

**Review by thanethomson — APPROVED** [review]
Thanks very much! :pray:

File: CHANGELOG.md (hunk 1 above)
Comment (thanethomson) [reply CHANGELOG.md]: Good point, I'll update that in the contributing guidelines :+1:

**Pull request merged**